tests/webTestServer.ts

```diff
@@ -640,31 +664,131 @@ function handleApiResolve(req: http.ServerRequest, res: http.ServerResponse) {
     });
 }
 
+function handleApiEnumerateTestFiles(req: http.ServerRequest, res: http.ServerResponse) {
+    readContent(req, (err, content) => {
+        try {
+            if (err) return sendInternalServerError(res, err);
+            if (!content) return sendBadRequest(res);
+            const tests: (string | FileTest)[] = enumerateTestFiles(content);
+            return sendJson(res, /*statusCode*/ 200, tests);
+        }
+        catch (e) {
+            return sendInternalServerError(res, e);
+        }
+    });
+}
+
+function enumerateTestFiles(runner: string) {
+    switch (runner) {
+        case "conformance":
+        case "compiler":
+            return listFiles(`tests/cases/${runner}`, /*serverDirname*/ undefined, /\.tsx?$/, { recursive: true }).map(parseCompilerTestConfigurations);
+        case "fourslash":
+            return listFiles(`tests/cases/fourslash`, /*serverDirname*/ undefined, /\.ts/i, { recursive: false });
+        case "fourslash-shims":
+            return listFiles(`tests/cases/fourslash/shims`, /*serverDirname*/ undefined, /\.ts/i, { recursive: false });
+        case "fourslash-shims-pp":
+            return listFiles(`tests/cases/fourslash/shims-pp`, /*serverDirname*/ undefined, /\.ts/i, { recursive: false });
+        case "fourslash-server":
+            return listFiles(`tests/cases/fourslash/server`, /*serverDirname*/ undefined, /\.ts/i, { recursive: false });
+        default:
+            throw new Error(`Runner '${runner}' not supported in browser tests.`);
+    }
+}
+
+// Regex for parsing options in the format "@Alpha: Value of any sort"
+const optionRegex = /^[\/]{2}\s*@(\w+)\s*:\s*([^\r\n]*)/gm;  // multiple matches on multiple lines
+
+function extractCompilerSettings(content: string): Record<string, any> {
+    const opts: Record<string, any> = {};
+
+    let match: RegExpExecArray;
+    /* tslint:disable:no-null-keyword */
+    while ((match = optionRegex.exec(content)) !== null) {

```

Comment:
Nit: prefer `// tslint:disable-line no-null-keyword` on this line, or `// tslint:disable-next-line no-null-keyword` on the previous line

Reply:
Actually, that's unneeded since I've disabled that rule throughout the whole file.